modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -0,0 +1,142 @@
+
+		if (implementedClassNames.contains("BaseModelListener")) {
+			return content;
+		}
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			String newJavaMethodContent = _checkMethodDefinition(
+				javaMethodContent, javaMethod, _onAfterUpdatePattern);
+
+			newJavaMethodContent = _checkMethodDefinition(
+				newJavaMethodContent, javaMethod, _onfBeforeUpdatePattern);
+
+			content = StringUtil.replace(
+				content, javaMethodContent, newJavaMethodContent);
+		}
+
+		return content;
+	}
+
+	private String _checkMethodDefinition(
+		String javaMethodContent, JavaMethod javaMethod, Pattern pattern) {
+
+		Matcher matcher = pattern.matcher(javaMethodContent);
+
+		if (matcher.find()) {
+			Matcher superMatcher = _superPattern.matcher(javaMethodContent);
+
+			if (superMatcher.find()) {
+				javaMethodContent = _checkSuper(
+					javaMethodContent, superMatcher.start());
+			}
+
+			JavaSignature javaSignature = javaMethod.getSignature();
+
+			List<JavaParameter> parameters = javaSignature.getParameters();
+
+			if (parameters.size() != 1) {
+				return javaMethodContent;
+			}
+
+			JavaParameter javaParameter = parameters.get(0);
+
+			String javaParameterName = javaParameter.getParameterName();
+
+			String javaParameterType = javaParameter.getParameterType();
+
+			String newParameter = StringBundler.concat(
+				javaParameterType, " original",
+				StringUtils.capitalize(javaParameterName));
+
+			String newParameters = StringBundler.concat(
+				newParameter, StringPool.COMMA_AND_SPACE, javaParameterType,
+				StringPool.SPACE, javaParameterName);
+
+			String methodCall = JavaSourceUtil.getMethodCall(
+				javaMethodContent, matcher.start());
+
+			javaMethodContent = StringUtil.replace(
+				javaMethodContent, JavaSourceUtil.getParameters(methodCall),
+				newParameters);
+		}
+
+		return javaMethodContent;
+	}
+
+	private String _checkSuper(String javaMethodContent, int pos) {
+		String methodCall = JavaSourceUtil.getMethodCall(
+			javaMethodContent, pos);
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
+
+		if (parameterList.size() != 1) {
+			return javaMethodContent;
+		}
+
+		String parameter = JavaSourceUtil.getParameters(methodCall);
+
+		String newParameters = StringBundler.concat(
+			"original", StringUtils.capitalize(parameter),
+			StringPool.COMMA_AND_SPACE, parameter);
+
+		String newMethodCall = StringUtil.replace(
+			methodCall, parameter, newParameters);
+
+		return StringUtil.replace(javaMethodContent, methodCall, newMethodCall);
+	}
+
+	private static final Pattern _onAfterUpdatePattern = Pattern.compile(
+		"onAfterUpdate\\s*\\(\\s*.+ .+\\)");
+	private static final Pattern _onfBeforeUpdatePattern = Pattern.compile(
+		"onBeforeUpdate\\s*\\(\\s*.+ .+\\)");
```

Comment:
@michaelmdelcavalcanti, we can merge these two patterns into one and remove the pattern parameter of the `_checkMethodDefinition` method.

```
private static final Pattern _onAfterOrBeforeUpdatePattern = Pattern.compile(
		"on(?:After|Before)Update\\s*\\(\\s*.+ .+\\)");
```